docs/integration/guidelines/kafka-guidelines.md

```diff
@@ -0,0 +1,84 @@
+
+Deletion of (published API) topics has to take the other direction, to ensure that no consumer loses a critical resource without prior notice. First of all, all subscribers (see previous section) have to cancel their subscription of the topic explicitly (which also means their associated ACLs are removed). This is an explicit statement that they do no longer rely on this resource. Once a topic in production stage does not have subscribers, it is eligible for deletion by the topic owner.
+
+Would the deletion be propagated the other way, consumers could lose their critical resource in production as soon as the staging is performed, although they may have already adapted in an integration stage. There would be no way for the topic owners to "know" when it is safe to stage the topic deletion to production.
+
+Of course, it seems impractical to get all consumers to "leave" the topic which the owner wants to delete. This is why topics shall be marked as deprecated as soon as a deletion is desired. The deprecation mark shall include an EOL date until which the topic is guaranteed to exist (the EOL date must, of course, give consumers enough time to adapt). Consumers shall be notified automatically about the deprecation of a topic, and after the EOL date, topic owners are allowed to delete the topic even if there are still subscribers.
+
+As a nice side effect, if consumers adapt faster than the EOL date, the topic still may be deleted before the EOL date if no more subscribers for the topic are registered.
+
+## Topic naming convention
+For Bane NOR we have defined the following guidelines for how to build a Topic Name. With this naming convention, we will have the flexibility in the future to support further domain and sub-domains, different event types, and to manage breaking changes (if absolutely needed) with versioning. The Topic name must contain the associated Business Capability and a descriptive name of the event, but not contain the name of the producing application(s).
+
+Environment - Describes the origin of data e.g. cloud or on-premises
+Domain - Describes the domain - for us in Bane NOR, we will have to learn what are top-level domains are
+Sub-domain - Describes the sub-domain. This will be optional, filled with default values, but in the future we see the need to define a more granular detail level with regards to events happening within one domain into sub-domains. 
+Topic type - Describes the type of topic, see above for details.
+Event - Describes the actual event, e.g. train-arrived-at-station
+Version - Describes the version of the topic. This will be used if there are any breaking changes on the topic, but it should rarely be used as we should do proper work with defining the required information related to events for the topic before we publish it.
+
+Example:
+```text
+
+{on-premises}.{operational}.{optional}.{event}.{train-arrived-at-station}.{v1}
```

Comment:
We talked about maybe changing name here in the example as "on-premesis" will probably not be the actual name of this. One solution is to provide example with "cloud" instead.